extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/supervisor/KinesisSupervisor.java

```diff
@@ -314,4 +356,94 @@ protected boolean useExclusiveStartSequenceNumberForNonFirstSequence()
   {
     return true;
   }
+
+  @Override
+  protected Map<String, OrderedSequenceNumber<String>> filterDeadShardsFromStartingOffsets(
+      Map<String, OrderedSequenceNumber<String>> startingOffsets
+  )
+  {
+    Map<String, OrderedSequenceNumber<String>> filteredOffsets = new HashMap<>();
+    for (Map.Entry<String, OrderedSequenceNumber<String>> entry : startingOffsets.entrySet()) {
+      if (!entry.getValue().get().equals(KinesisSequenceNumber.END_OF_SHARD_MARKER)) {
+        filteredOffsets.put(entry.getKey(), entry.getValue());
+      } else {
+        log.info("Excluding shard[%s] because it has reached EOS.", entry.getKey());
```

Comment:
A thought: double-check (if you haven't already) that this log message doesn't get logged too often.

Reply:
Dropped this to a debug log